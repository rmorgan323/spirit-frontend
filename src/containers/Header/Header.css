/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 16px;
  font-family: "Lato", sans-serif;
  outline: none;
  color: #6d6e71; }

h1, h2, h3, h4, h5, h6 {
  font-family: "Josefin Sans", sans-serif;
  line-height: 1.3; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 30px; }

h3 {
  font-size: 24px; }

h4 {
  font-size: 18px; }

h5 {
  font-size: 14px; }

h6 {
  font-size: 10px; }

p {
  font-size: 14px;
  line-height: 1.3;
  font-weight: 900; }

::placeholder {
  opacity: .5;
  text-align: center; }

/***  MIXINS  ***/
.Header {
  background-color: #BFAFD0;
  padding: 7px 20px;
  border-bottom: 5px solid #7F60A4; }
  .Header .header-wordmark {
    margin: 0;
    width: 150px; }
  .Header .header-welcome-user, .Header .login-link {
    position: absolute;
    right: 20px;
    top: -9px;
    color: #F9F9F9;
    font-weight: 400;
    text-decoration: none;
    transition: 0.3s ease-in-out; }
  .Header .login-link:hover {
    border-bottom: 1px solid #F9F9F9;
    cursor: pointer; }
  .Header .header-welcome-clinic {
    position: absolute;
    right: 20px;
    top: 15px;
    color: #F9F9F9;
    font-weight: 400; }
