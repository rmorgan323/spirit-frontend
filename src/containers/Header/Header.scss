@import '../../variables.scss';

.Header {
  background-color: $purple3;
  padding: 7px 20px;
  border-bottom: 5px solid $purple5;

  .header-wordmark {
    margin: 0;
    width: 150px;
  }

  .login-link {
    position: absolute;
    right: 20px;
    top: -9px;
    color: $white1;
    font-weight: 400;
    text-decoration: none;
    transition: 0.3s ease-in-out;
  }

  .login-link:hover {
    border-bottom: 1px solid $white1;
    cursor: pointer;
  }

  .header-welcome-clinic {
    position: absolute;
    right: 20px;
    top: 15px;
    color: $white1;
    font-weight: 400;
  }

  .nav-icon {
    display: flex;
    flex-direction: column;
    position: absolute;
    top: 10px;
    right: 10px;
    z-index: 3;
    &.close {
      width: 30px;
      height: 30px;
    }
  }
  .burger-line {
    height: 4px;
    width: 30px;
    margin: 3px;
    background-color: $purple6;
    border-radius: 2px;
    transition: all 0.25s;
    &.close {
      background-color: $white1;
      position: absolute;
      top: 10px;
    }
    &.one {
      transform: rotate(135deg);
    }
    &.two {
      transform: rotate(-135deg);
    }
    &.remove {
      display: none;
    }
  }
  .navigation {
    position: absolute;
    top: 0;
    right: 0;
    display: flex;
    flex-direction: column;
    width: 250px;
    z-index: 2;
  }

  .header-welcome-user {
    margin: 0;
    padding: 15px 10px;
    color: $white1;
    background-color: $purple7;
    box-shadow: rgba(0,0,0,0.3) -3px 3px 5px;
    font-weight: 400;
  }

  .nav-link {
    padding: 10px;
    box-shadow: rgba(0,0,0,0.3) -3px 3px 5px;
    color: $white1;
    background-color: $purple3;
    text-decoration: none;
    text-transform: capitalize;
    transition: 0.2s ease-in-out;
    &:hover {
      background-color: $purple4;
    }
    &.selected {
      background-color: $purple5;
      cursor: default;
      pointer-events: none;
    }
  }
}