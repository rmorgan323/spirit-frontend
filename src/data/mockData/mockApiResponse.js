const mockApiResponse = {
  status: 'success'
};

export default mockApiResponse;