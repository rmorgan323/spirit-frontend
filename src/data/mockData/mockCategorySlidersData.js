/*eslint-disable camelcase*/

export const mockTitle = 'Capacities';

export const mockSlidersArray = [
  'Attention',
  'Action',
  'Affect',
  'Autonomic',
  'Arousability'
];

export const mockDatabaseNamesArray = [
  'mod_2_attention',
  'mod_2_action',
  'mod_2_affect',
  'mod_2_autonomic',
  'mod_2_arousability'
];

export const mockReminderAsterisk = [null, '*', '*', '*', '*'];

export const mockReminder = '* If selected, remember to make a hypothesized arousal chart.';

export const mockSliderTitle = 'Attention';

export const mockDatabaseName = 'mod_2_attention';

