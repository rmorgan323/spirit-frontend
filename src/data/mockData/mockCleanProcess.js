/*eslint-disable camelcase*/

export const mockCleanProcess = {
  id: 10,
  sen_h_vestibular: null,
  sen_h_proprioception: null,
  sen_h_tactile: null,
  sen_h_auditory: null,
  sen_h_visual: null,
  sen_h_intero: null,
  sen_1_ability: null,
  sen_1_match: null,
  sen_1_classification: null,
  sen_1_spatial: null,
  sen_1_figure: null,
  sen_1_inferences: null,
  sen_2_initiation: null,
  sen_2_imitation: null,
  sen_2_affordance: null,
  sen_2_generativity: null,
  sen_3_sequencing: null,
  sen_3_sequencing_gestalt: null,
  sen_3_schema_selection: null,
  sen_3_schema_expansion: null,
  sen_3_modification: null,
  sen_4_timing: null,
  sen_4_timing_force: null,
  sen_4_spatial: null,
  sen_4_temporal: null,
  sen_4_control: null,
  sen_4_monitoring: null,
  sen_5_oral: null,
  sen_5_ocular: null,
  sen_5_postural: null,
  sen_5_limb: null,
  sen_5_gross: null,
  sen_5_fine: null,
  sen_5_tool: null,
  sen_5_projected: null,
  sen_6_somatodyspraxia: null,
  sen_6_visual: null,
  sen_6_generalized: null,
  sen_6_bmc: null,
  sen_6_praxis: null,
  mod_h_vestibular: null,
  mod_h_proprioception: null,
  mod_h_tactile: null,
  mod_h_auditory: null,
  mod_h_visual: null,
  mod_h_intero: null,
  mod_1_defensiveness: null,
  mod_1_over: null,
  mod_1_under: null,
  mod_1_variable: null,
  mod_2_attention: null,
  mod_2_action: null,
  mod_2_affect: null,
  mod_2_autonomic: null,
  mod_3_physiological: null,
  mod_3_sensory: null,
  mod_3_motor: null,
  mod_3_emotional: null,
  mod_3_cognitive: null,
  mod_3_social: null,
  mod_3_prosocial: null,
  mod_4a_passive: null,
  mod_4a_co_regulation: null,
  mod_4a_approach: null,
  mod_4a_adaptive: null,
  mod_4b_active: null,
  mod_4b_self_regulation: null,
  mod_4b_avoid: null,
  mod_4b_maladaptive: null,
  exe_1_orient: null,
  exe_1_inhibition: null,
  exe_2a_working: null,
  exe_2a_shifting: null,
  exe_2b_verbal: null,
  exe_2b_spatial: null,
  exe_2b_general: null,
  exe_2b_global: null,
  exe_2b_flexibility: null,
  exe_3_sequencing: null,
  exe_3_monitoring: null,
  exe_4a_future: null,
  exe_4a_spatial: null,
  exe_4a_problem: null,
  exe_4a_reflective: null,
  exe_4a_internalized: null,
  exe_4b_motivational: null,
  exe_4b_effortful: null,
  exe_4b_self_control: null,
  exe_4b_self_regulation: null,
  exe_4b_self_compassion: null,
  pos_h_vestibular: null,
  pos_h_proprioception: null,
  pos_h_tactile: null,
  pos_h_auditory: null,
  pos_h_visual: null,
  pos_h_intero: null,
  pos_2_stability: null,
  pos_2_mobility: null,
  pos_2_tonic: null,
  pos_2_phasic: null,
  pos_2_muscle: null,
  pos_3_sagittal: null,
  pos_3_frontal: null,
  pos_3_transverse: null,
  pos_3_midlines: null,
  pos_5_suck: null,
  pos_5_ocular: null,
  pos_5_antigravity: null,
  pos_5_reflex: null,
  pos_5_alignment_COG: null,
  pos_5_alignment_orient: null,
  pos_5_anticipatory: null,
  pos_5_interactive: null,
  pos_6_push: null,
  pos_6_bilateral: null,
  pos_6_grasp: null,
  pos_6_strength: null,
  pos_6_effortful: null,
  soc_1_regulation: null,
  soc_1_engagement: null,
  soc_1_reciprocity: null,
  soc_1_problem: null,
  soc_1_representation: null,
  soc_1_triangular: null,
  soc_1_comparative: null,
  soc_1_reflective: null,
  soc_2_self_awareness: null,
  soc_2_joint: null,
  soc_2_social_referencing: null,
  soc_2_imitation: null,
  soc_2_inferential: null,
  soc_2_contingent: null,
  soc_2_theory: null,
  soc_2_central: null,
  soc_2_mentalizing: null,
  soc_2_empathy: null,
  soc_2_social_motivators: null,
  soc_2_pro_social: null,
  soc_3_affective: null,
  soc_3_emotional_contagion: null,
  soc_3_activation_low: null,
  soc_3_emotional_cascade: null,
  soc_3_activation_high: null,
  soc_3_emotional_differentiation: null,
  soc_3_emotional_association: null,
  soc_3_self_control: null,
  soc_3_self_regulation: null,
  soc_3_self_compassion: null,
  soc_4_mental: null,
  soc_4_thoughts: null,
  soc_4_attachments: null,
  soc_4_trauma: null,
  session_id: 10,
  created_at: "2018-02-20T21:25:45.454Z",
  updated_at: "2018-02-20T21:25:45.454Z",
  mod_2_arousability: null,
  exe_3_initiation: null,
  exe_3_planning: null,
  pos_4_core: null,
  pos_4_shoulder: null,
  pos_4_pelvic: null,
  pos_4_head: null,
  pos_4_eyes: null,
  pos_4_hand: null,
  pos_4_lower: null,
  pos_4_foot: null,
  soc_2_social_perspective: null,
  pos_1_outer: null,
  pos_1_inner: null,
  exe_4a_generativity: null
};
