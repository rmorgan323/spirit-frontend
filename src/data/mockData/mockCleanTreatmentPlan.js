/*eslint-disable camelcase*/

export const mockCleanTreatmentPlan = {
  id: 10,
  modulation_sensory: null,
  modulation_task: null,
  modulation_environment: null,
  modulation_predictability: null,
  modulation_self_regulation: null,
  modulation_interaction: null,
  modulation_JRC_AR_notes: null,
  posture_sensory: null,
  posture_task: null,
  posture_environment: null,
  posture_predictability: null,
  posture_self_regulation: null,
  posture_interaction: null,
  posture_JRC_AR_notes: null,
  sensory_discrimination_sensory: null,
  sensory_discrimination_task: null,
  sensory_discrimination_environment: null,
  sensory_discrimination_predictability: null,
  sensory_discrimination_self_regulation: null,
  sensory_discrimination_interaction: null,
  sensory_discrimination_JRC_AR_notes: null,
  social_emotional_sensory: null,
  social_emotional_task: null,
  social_emotional_environment: null,
  social_emotional_predictability: null,
  social_emotional_self_regulation: null,
  social_emotional_interaction: null,
  social_emotional_JRC_AR_notes: null,
  executive_functioning_sensory: null,
  executive_functioning_task: null,
  executive_functioning_environment: null,
  executive_functioning_predictability: null,
  executive_functioning_self_regulation: null,
  executive_functioning_interaction: null,
  executive_functioning_JRC_AR_notes: null,
  session_id: 10,
  created_at: "2018-02-20T21:25:45.468Z",
  updated_at: "2018-02-20T21:25:45.468Z"
};
