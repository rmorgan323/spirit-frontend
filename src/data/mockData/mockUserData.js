/*eslint-disable camelcase*/

export const mockUserData = {
  id: 1,
  authrocket_id: "usr_0vBdVVz3RUsbR2ArDCoPvo",
  name: "Lee Trevino",
  email: "sample@samplemail.com",
  admin: false,
  created_at: "2018-02-19T03:41:23.701Z",
  updated_at: "2018-02-19T03:41:23.701Z",
  clinic: "I Need Therapy",
  clinic_abbreviation: "INT",
  clinic_id: 10,
  clinic_passcode: "utk85fmf"
};
