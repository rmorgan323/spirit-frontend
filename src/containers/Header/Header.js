import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import authLogin from '../../helpers/authLogin';
import { PropTypes } from 'prop-types';
import './Header.css';

export const Header = props => {
  const { name, id, clinic } = props.user;

  return (
    <div className="Header">
      <Link to="/spirit/index">
        <img
          className="header-wordmark"
          src="/assets/Spirit-Wordmark-White.png"
          alt="SpIRiT-wordmark"
        />
      </Link>

      {name && (
        <Fragment>
          <Link to={`/spirit/users/${id}`}>
            <h5 className="header-welcome-user">Welcome, {name}!</h5>
          </Link>
          {clinic && <h5 className="header-welcome-clinic">
            {`Clinic: ${clinic}`}
          </h5>}
        </Fragment>
      )}

      {!name && (
        <Fragment>
          <a href={authLogin}>
            <h5 className="login-link">LOGIN</h5>
          </a>
        </Fragment>
      )}
    </div>
  );
};

export const mapStateToProps = store => ({
  user: store.user
});

export default connect(
  mapStateToProps,
  null
)(Header);

Header.propTypes = {
  user: PropTypes.object
};
