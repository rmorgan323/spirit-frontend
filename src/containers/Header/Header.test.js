import React from 'react';
import { Header, mapStateToProps } from './Header';
import { shallow } from 'enzyme';
import { mockUserData } from '../../data/mockData/mockUserData';

describe('Header tests', () => {
  let user;
  let renderedHeader;

  beforeEach(() => {
    user = mockUserData;

    renderedHeader = shallow(<Header user={user} />);
  });

  it('Should match the snapshot', () => {
    expect(renderedHeader).toMatchSnapshot();
  });
});

describe('mapStateToProps tests', () => {
  it('Should pull user from store', () => {
    const user = mockUserData;
    const mockStore = { user };
    const result = mapStateToProps(mockStore);

    expect(result.user).toEqual(mockStore.user);
  });
});
