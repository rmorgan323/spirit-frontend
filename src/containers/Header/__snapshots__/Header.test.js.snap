// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    user={
      Object {
        "admin": false,
        "authrocket_id": "usr_0vBdVVz3RUsbR2ArDCoPvo",
        "clinic": "I Need Therapy",
        "clinic_abbreviation": "INT",
        "clinic_id": 10,
        "clinic_passcode": "utk85fmf",
        "created_at": "2018-02-19T03:41:23.701Z",
        "email": "sample@samplemail.com",
        "id": 1,
        "name": "Lee Trevino",
        "updated_at": "2018-02-19T03:41:23.701Z",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/spirit/index"
        >
          <img
            alt="SpIRiT-wordmark"
            className="header-wordmark"
            src="/assets/Spirit-Wordmark-White.png"
          />
        </Link>,
        <UNDEFINED>
          <Link
            replace={false}
            to="/spirit/users/1"
          >
            <h5
              className="header-welcome-user"
            >
              Welcome, 
              Lee Trevino
              !
            </h5>
          </Link>
          <h5
            className="header-welcome-clinic"
          >
            Clinic: I Need Therapy
          </h5>
        </UNDEFINED>,
        false,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            alt="SpIRiT-wordmark"
            className="header-wordmark"
            src="/assets/Spirit-Wordmark-White.png"
          />,
          "replace": false,
          "to": "/spirit/index",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "SpIRiT-wordmark",
            "className": "header-wordmark",
            "src": "/assets/Spirit-Wordmark-White.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/spirit/users/1"
            >
              <h5
                className="header-welcome-user"
              >
                Welcome, 
                Lee Trevino
                !
              </h5>
            </Link>,
            <h5
              className="header-welcome-clinic"
            >
              Clinic: I Need Therapy
            </h5>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h5
                className="header-welcome-user"
              >
                Welcome, 
                Lee Trevino
                !
              </h5>,
              "replace": false,
              "to": "/spirit/users/1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Welcome, ",
                  "Lee Trevino",
                  "!",
                ],
                "className": "header-welcome-user",
              },
              "ref": null,
              "rendered": Array [
                "Welcome, ",
                "Lee Trevino",
                "!",
              ],
              "type": "h5",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Clinic: I Need Therapy",
              "className": "header-welcome-clinic",
            },
            "ref": null,
            "rendered": "Clinic: I Need Therapy",
            "type": "h5",
          },
        ],
        "type": Symbol(react.fragment),
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/spirit/index"
          >
            <img
              alt="SpIRiT-wordmark"
              className="header-wordmark"
              src="/assets/Spirit-Wordmark-White.png"
            />
          </Link>,
          <UNDEFINED>
            <Link
              replace={false}
              to="/spirit/users/1"
            >
              <h5
                className="header-welcome-user"
              >
                Welcome, 
                Lee Trevino
                !
              </h5>
            </Link>
            <h5
              className="header-welcome-clinic"
            >
              Clinic: I Need Therapy
            </h5>
          </UNDEFINED>,
          false,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="SpIRiT-wordmark"
              className="header-wordmark"
              src="/assets/Spirit-Wordmark-White.png"
            />,
            "replace": false,
            "to": "/spirit/index",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "SpIRiT-wordmark",
              "className": "header-wordmark",
              "src": "/assets/Spirit-Wordmark-White.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/spirit/users/1"
              >
                <h5
                  className="header-welcome-user"
                >
                  Welcome, 
                  Lee Trevino
                  !
                </h5>
              </Link>,
              <h5
                className="header-welcome-clinic"
              >
                Clinic: I Need Therapy
              </h5>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <h5
                  className="header-welcome-user"
                >
                  Welcome, 
                  Lee Trevino
                  !
                </h5>,
                "replace": false,
                "to": "/spirit/users/1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Welcome, ",
                    "Lee Trevino",
                    "!",
                  ],
                  "className": "header-welcome-user",
                },
                "ref": null,
                "rendered": Array [
                  "Welcome, ",
                  "Lee Trevino",
                  "!",
                ],
                "type": "h5",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Clinic: I Need Therapy",
                "className": "header-welcome-clinic",
              },
              "ref": null,
              "rendered": "Clinic: I Need Therapy",
              "type": "h5",
            },
          ],
          "type": Symbol(react.fragment),
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
